### PR TITLE
UI Base - Update how we configure client data inclusion and injection of "auth" plugins

### LINKED ISSUE #853: Allow for socketId constraint on all widget types
### Description

With the FlowFuse Authentication plugin we introduced a sidebar which allowed users to control which widget types can be constrained to a single connection/user. This is a popular feature, but not entirely bound to the concept of authentication with FlowFuse.

We've had it requested multiple times to allow for `msg._client.socketId` to be used in order to constrain messages to partiular connections-only. This is already possible for `ui-control` and `ui-event` nodes, but only these nodes in the pure open-source version of Dashboard 2.0

This also makes way for future auth plugins (that aren't FlowFuse) to utilise the same configuration of which widgets allow constraints

### Linked Issues

- https://github.com/FlowFuse/node-red-dashboard/issues/415
- https://github.com/FlowFuse/node-red-dashboard/issues/430
- https://github.com/FlowFuse/node-red-dashboard/issues/638

### Have you provided an initial effort estimate for this issue?

I have provided an initial effort estimate

### PR DESCRIPTION
**Pull request by joepavitt**
## Description

- Migrate the client constraints/data configuration sidebar from the FF Auth plugin to core
- Introduces socket constraints/control to _all_ widget types, including third-party widgets
- Introduce a "client data providers" list in the sidebar to make it clear what data is appending to `msg._client`
- Update documentation for plugins
- Add documentation for how to use the client configuration (and update the link in the Editor)

## Related Issue(s)

Closes #853 
Related to https://github.com/FlowFuse/node-red-dashboard-ff-auth/pull/14

### CODE REVIEW

**Comment by joepavitt** [comment]
Screenshot of the sidebar:

File: nodes/config/ui_base.html [hunk 1]
```diff
@@ -1467,6 +1679,10 @@
                         buildLayoutOrderEditor(pagesContent)
                         // add Themes View
                         buildThemesEditor(themesContent)
+                        // add Themes View
+                        console.log('sidebar on change')
```
Comment (ZJvandeWeg) [inline nodes/config/ui_base.html:1683]: 🙈

Comment (joepavitt) [reply nodes/config/ui_base.html:1683]: It's in Draft - hadn't done a proper sweep myself yet - just wanted to get something pushed up to round the day off

File: docs/user/sidebar.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+<script setup>
+    import AddedIn from '../components/AddedIn.vue';
+</script>
+
+# Dashboard 2.0 Sidebar
+
+![Screenshot showing the Dashboard 2.0 sidebar in the Node-RED Editor.](/images/dashboard-sidebar.png){data-zoomable}
+_Screenshot showing the Dashboard 2.0 sidebar in the Node-RED Editor._
+
+The Dashboard 2.0 sidebar is a menu added to the right-hand side of the Node-RED Editor. It is designed to provide a quick overview of the current state of the Dashboard, and support in the overall configuration of the Dashboard.
```
Comment (Steve-Mcl) [inline docs/user/sidebar.md:11]:
```suggestion
The Dashboard 2.0 sidebar tab is added to the Node-RED sidebar. It is designed to provide a quick overview of the current state of the Dashboard, and support in the overall configuration of the Dashboard.
```

**Pull request merged**